virtcontainers/qemu_s390x.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2018 IBM
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	govmmQemu "github.com/intel/govmm/qemu"
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+	"github.com/sirupsen/logrus"
+)
+
+type qemuS390x struct {
+	// inherit from qemuArchBase, overwrite methods if needed
+	qemuArchBase
+}
+
+const defaultQemuPath = "/usr/bin/qemu-system-s390x"
+
+const defaultQemuMachineType = QemuCCWVirtio
+
+const defaultQemuMachineOptions = "accel=kvm"
+
+const defaultPCBridgeBus = "pci.0"
+
+const VirtioSerialCCW = "virtio-serial-ccw"
+
+var qemuPaths = map[string]string{
+	QemuCCWVirtio: defaultQemuPath,
+}
+
+var kernelRootParams = []Param{}
+
+// Verify needed parameters
+var kernelParams = []Param{
+	{"console", "ttysclp0"},
+}
+
+var supportedQemuMachines = []govmmQemu.Machine{
+	{
+		Type:    QemuCCWVirtio,
+		Options: defaultQemuMachineOptions,
+	},
+}
+
+// MaxQemuVCPUs returns the maximum number of vCPUs supported
+func MaxQemuVCPUs() uint32 {
+	// Max number of virtual Cpu defined in qemu. See
+	// https://github.com/qemu/qemu/blob/80422b00196a7af4c6efb628fae0ad8b644e98af/target/s390x/cpu.h#L55
+	// #define S390_MAX_CPUS 248
```

Comment:
It's a shame this can't be queried by calling qemu (via govmm) directly. But clearly not a blocker as other architectures also hard-code a value.

Aside: For the future, I do wonder if we could push the hard-coded arch-specific values down into govmm rather than having them in the runtime though - wdyt @markdryan?

Reply:
Sure - this was more a question for @markdryan ;)